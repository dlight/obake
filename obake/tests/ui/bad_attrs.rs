#[obake::versioned]
#[obake(version("0.1.0", extra_nonsense))]
struct Foo {}

#[obake::versioned]
#[obake(version("0.1.0"))]
struct Bar {
    #[obake(cfg("*"), extra_nonsense)]
    field_0: u32,
}

#[obake::versioned]
#[obake(version("0.1.0"))]
enum Baz {
    Variant {
        #[obake(not_an_obake_helper)]
        field_0: u32,
    },
}

#[obake::versioned]
#[obake(version("0.1.0"))]
enum Flim {
    Variant {
        #[obake(inherit, extra_nonsense)]
        field_0: u32,
    },
}

#[obake::versioned]
#[obake(version("0.1.0"))]
enum Flam {
    Variant {
        #[obake(cfg("*", extra_nonsense))]
        field_0: u32,
    },
}

fn main() {}
