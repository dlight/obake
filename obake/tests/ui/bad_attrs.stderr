error: unexpected token
 --> $DIR/bad_attrs.rs:2:24
  |
2 | #[obake(version("0.1.0", extra_nonsense))]
  |                        ^

error: unexpected token
 --> $DIR/bad_attrs.rs:8:21
  |
8 |     #[obake(cfg("*"), extra_nonsense)]
  |                     ^

error: unrecognised `obake` helper attribute
  --> $DIR/bad_attrs.rs:16:17
   |
16 |         #[obake(not_an_obake_helper)]
   |                 ^^^^^^^^^^^^^^^^^^^

error: unexpected token
  --> $DIR/bad_attrs.rs:25:24
   |
25 |         #[obake(inherit, extra_nonsense)]
   |                        ^

error: unexpected token
  --> $DIR/bad_attrs.rs:34:24
   |
34 |         #[obake(cfg("*", extra_nonsense))]
   |                        ^
